tests/handlers/koji/test_rebuild_images_on_rpm_advisory_change.py

```diff
@@ -664,8 +672,57 @@ def test_affected_packages_with_modules(self, exists, affected_rpm_nvrs):
             release_categories=conf.container_release_categories,
             leaf_container_images=None,
             skip_nvrs=None,
+            restrict_to_ext_repos=None,
+        )
+
+    @patch(
+        "freshmaker.handlers.koji.RebuildImagesOnRPMAdvisoryChange._lookup_external_repos",
+        return_value = ["foo/bar/repo"]
+    )
+    def test_moderate_cve_external_repo(self, mock_lookup):
+        self.event.security_impact = "moderate"
+        self.event.advisory.is_compliance_priority = True
+        
+        self.handler._find_images_to_rebuild(123456)
+        self.find_images_to_rebuild.assert_called_once_with(
+            ["httpd-2.4-11.el7"],
+            ["content-set-1", "pulp_repo_x86_64"],
+            filter_fnc=self.handler._filter_out_not_allowed_builds,
+            published=None,
+            release_categories=None,  #TODO check if this is okay
+            leaf_container_images=None,
+            skip_nvrs=None,
+            restrict_to_ext_repos=["foo/bar/repo"],
         )
 
+    @patch(
+        "freshmaker.handlers.koji.RebuildImagesOnRPMAdvisoryChange._lookup_external_repos",
+        side_effect = HTTPError(f"404 Client Error: not found for url: foo.bar/baz")
+    )
+    def test_moderate_cve_external_repo_http_error(self, mock_lookup):
+        self.event.security_impact = "moderate"
+        self.event.advisory.is_compliance_priority = True
+        
+        result = self.handler._find_images_to_rebuild(123456)
+
+        db_event = Event.get_or_create_from_event(db.session, self.event)
+        self.assertEqual(db_event.state, 4)
+        self.assertEqual(result, [])
+
+    @patch(
+        "freshmaker.handlers.koji.RebuildImagesOnRPMAdvisoryChange._lookup_external_repos",
+        return_value = []
+    )
+    def test_moderate_cve_external_repo_empty(self, mock_lookup):
```

Comment:
This and the previous test are pretty similar. You could try to use `parametrize`. It allows you to change the variable values, but keep the code as it is. You can see an example in `tests/test_utils.py`. It's not always possible to use it (depends on how the test is written), but I think it should work in this case. At least, give it a try.

Reply:
I didn't know about this trick in pytest, that is so cool!

I gave it a try, and I kept getting a "missing argument" error. It seems that parametrize is not compatible with `unittest.TestCase` subclasses according to [this](https://docs.pytest.org/en/stable/how-to/unittest.html#pytest-features-in-unittest-testcase-subclasses).

However, I saw that there is a llib called [parameterized](https://github.com/wolever/parameterized) that solves just that! I would be happy to try it out, if you think it is a nice addition to the project. What do you think, @gnaponie and @qixiang?

Reply:
I don't have a strong preference about `parametrize` or `parametrized`. I've also incurred some issues with `parametrize` from pytest in the past (I think it might be incompatible with some other decorator or something). So you can try parametrized if it is easier.

Reply:
Alright. Let's skip it then. I believe the change looks good at this point :+1: 
good job